src/compiler/core.ts

```diff
@@ -942,15 +1030,26 @@ namespace ts {
      * Adds the value to an array of values associated with the key, and returns the array.
      * Creates the array if it does not already exist.
      */
-    export function multiMapAdd<V>(map: Map<V[]>, key: string | number, value: V): V[] {
-        const values = map[key];
+    export function multiMapAdd<V>(map: Map<V[]>, key: string, value: V): V[] {
+        let values = map.get(key);
+        if (values) {
+            values.push(value);
+        }
+        else {
+            map.set(key, values = [value]);
+        }
+        return values;
+    }
+
+    export function multiMapSparseArrayAdd<V>(map: SparseArray<V[]>, key: number, value: V): V[] {

```

Comment:
This is hard to do since we return the value as a SparseArray, so we'd have to add `as any as string` at all the external uses too. At least I've moved this function to `factory.ts`.